cmd/celestia-appd/cmd/pre_start_hooks.go

```diff
@@ -0,0 +1,31 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+type PreStartHook func(cmd *cobra.Command) error
+
+// addPreStartHooks finds the start command and adds pre-start hooks using Cobra's PreRunE
+func addPreStartHooks(rootCmd *cobra.Command, hooks ...PreStartHook) error {
```

Comment:
not blocking opinion

imo would be slightly more explicit if we could make the point that we're modifying things clearer

so simply returning the thing we're modifying even though we don't need to due to them being pointers

Reply:
I was leaning on the word "add" to imply that it was modifying the command. I can also return the command struct but I'm not sure what use that would be